File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        // neetcode solution O(n) time complexity O(1) space complexity
        int res = 0;
        int l = 0;
        int r = 1;

        while (r < prices.size())
        {
            if (prices[l] < prices[r])
                res = max(res, (prices[r] - prices[l]));
            else
                l = r;
            r++;
        }
        return res;

    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        # neetcode solution O(n) time complexity O(1) space complexity
        res = 0
        l, r = 0, 1
        while(r < (len(prices))):
            
            if prices[l] < prices[r]:
                res = max(res, (prices[r] - prices[l]))
            else:
                l = r
            r += 1
        return res
        